src/planner/Query.h

```diff
@@ -439,10 +439,105 @@ class GetEdges final : public Explore {
  */
 class IndexScan final : public Explore {
 public:
-    IndexScan(ExecutionPlan* plan, PlanNode* input, GraphSpaceID space)
-        : Explore(plan, Kind::kIndexScan, input, space) {}
+    using IndexQueryCtx = std::unique_ptr<std::vector<storage::cpp2::IndexQueryContext>>;
+    using IndexReturnCols = std::unique_ptr<std::vector<std::string>>;
+
+    static IndexScan* make(ExecutionPlan* plan, PlanNode* input, GraphSpaceID space) {
+        return new IndexScan(plan, input, space);
+    }
+
+    static IndexScan* make(ExecutionPlan* plan,
+                           PlanNode* input,
+                           GraphSpaceID space,
+                           IndexQueryCtx&& contexts,
+                           IndexReturnCols&& returnCols,
```

Comment:
Is this duplicate with the field colNames_?

Reply:
> Is this duplicate with the field colNames_?

Maybe yes. Let's leave it at that for now, whether change is needed depends on the validator.